This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.


From 9e5ebf550f2f678f52c8c42297fcc934682e5efa Mon Sep 17 00:00:00 2001
From: Boris Pek <tehnick-8@yandex.ru>
Date: Thu, 25 Oct 2018 03:33:36 +0300
Subject: [PATCH 1/1] Add support of cross-compilation for MS Windows

using MinGW and project MXE (https://mxe.cc/).


diff --git a/cgi-fcgi/Makefile.am b/cgi-fcgi/Makefile.am
index 19e74fc..0150322 100644
--- a/cgi-fcgi/Makefile.am
+++ b/cgi-fcgi/Makefile.am
@@ -11,5 +11,5 @@ INCLUDE_FILES   = $(INCLUDEDIR)/fastcgi.h  \
 LIBDIR      = ../libfcgi
 LIBFCGI     = $(LIBDIR)/libfcgi.la
 
-LDADD = $(LIBFCGI)
+cgi_fcgi_LDADD = $(LIBFCGI)
 cgi_fcgi_SOURCES = $(INCLUDE_FILES) cgi-fcgi.c
diff --git a/configure.in b/configure.in
index c5fbc4b..8498b6c 100755
--- a/configure.in
+++ b/configure.in
@@ -4,14 +4,16 @@ dnl     This file is an input file used by the GNU "autoconf" program to
 dnl     generate the file "configure", which is run during the build
 dnl     to configure the system for the local environment.
 
-AC_INIT(fcgi, 2.4.1-SNAP-0910052249)
+AC_INIT([fcgi], [2.4.2])
 AM_INIT_AUTOMAKE([1.11 foreign])
-
+AC_CONFIG_MACRO_DIR([m4])
 AM_CONFIG_HEADER(fcgi_config.h)
 
+LT_INIT([win32-dll])
+
 AC_PROG_CC
-AC_PROG_CPP 
-AC_PROG_INSTALL 
+AC_PROG_CPP
+AC_PROG_INSTALL
 AC_PROG_LIBTOOL
 
 AC_PROG_CXX
@@ -57,15 +59,20 @@ AC_REPLACE_FUNCS([strerror])
 
 AC_C_INLINE
 
-#--------------------------------------------------------------------
-#  This is a little hokie in that it avoids including config.guess
-#  and config.sub in the distribution, but its been working so far.
-#  Windows builds don't run configure so we should be safe fixing
-#  this to 'unix' (at least for now).
-#--------------------------------------------------------------------
-SYSTEM=unix
+AC_CANONICAL_HOST
+
+AS_CASE([$host_os],
+  [*mingw*], [SYSTEM=win32],
+  [SYSTEM=unix]
+)
 AC_SUBST([SYSTEM])
 
+AS_CASE([$host_os],
+  [*mingw*], [EXTRA_LIBS=-lws2_32],
+  [EXTRA_LIBS=]
+)
+AC_SUBST([EXTRA_LIBS])
+
 AC_PROG_CC_WARNINGS
 
 AC_ARG_WITH([pkgconfigdir],
@@ -83,3 +90,4 @@ AC_CONFIG_FILES([Makefile
                  fcgi++.pc])
 
 AC_OUTPUT
+
diff --git a/examples/log-dump.c b/examples/log-dump.c
index 0cac827..a583980 100644
--- a/examples/log-dump.c
+++ b/examples/log-dump.c
@@ -51,7 +51,9 @@ int main(void)
     while(FCGI_Accept() >= 0) {
         rolePtr = getenv("FCGI_ROLE");
 	if(rolePtr == NULL) {
+#ifndef _WIN32
 	    kill(getpid(), SIGQUIT);
+#endif
 	    exit(-1);
 	}
 	if(strstr(rolePtr, "AUTHORIZER")) {
diff --git a/libfcgi/Makefile.am b/libfcgi/Makefile.am
index a91b223..12e3331 100644
--- a/libfcgi/Makefile.am
+++ b/libfcgi/Makefile.am
@@ -18,10 +18,11 @@ libfcgi_la_SOURCES = $(INCLUDE_FILES)  \
                      os_@SYSTEM@.c
 libfcgi_la_CC      = @PTHREAD_CC@
 libfcgi_la_CFLAGS  = @PTHREAD_CFLAGS@
+libfcgi_la_LDFLAGS = @EXTRA_LIBS@ -no-undefined
 
 libfcgi___la_SOURCES = $(INCLUDE_FILES)       \
                        $(INCLUDEDIR)/fcgio.h  \
                        fcgio.cpp
 libfcgi___la_CFLAGS  = @PTHREAD_CFLAGS@
-libfcgi___la_LDFLAGS = -lfcgi -rpath @libdir@
-
+libfcgi___la_LIBADD  = libfcgi.la
+libfcgi___la_LDFLAGS = -rpath @libdir@ -no-undefined

